apps/jsonrpc2/lib/jsonrpc2/request.ex

```diff
@@ -0,0 +1,41 @@
+defmodule JSONRPC2.Request do
+  @moduledoc """
+  JSON-RPC 2.0 Request object utilites.
+  """
+
+  @type request ::
+          {JSONRPC2.method(), JSONRPC2.params()}
+          | {JSONRPC2.method(), JSONRPC2.params(), JSONRPC2.id()}
+
+  @doc """
+  Returns a serialized `request` using `serializer`.
+
+  See the README for more information on serializers.
+  """
+  @spec serialized_request(request, module) :: {:ok, binary()} | {:error, any}
+  def serialized_request(request, serializer) do
+    serializer.encode(request(request))
+  end
+
+  @doc """
+  Returns a map representing the JSON-RPC2.0 request object for `request`.
+  """
```

Comment:
Same here with examples, since the function is pretty straight-forward but the typespec doesn't give us many clues.